experimental/auth-logic/wrapper_interface_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authlogic contains logic and tests for interfacing with the
+// authorization logic compiler
+package authlogic
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+)
+
+type IntPair struct {
+	x int
+	y int
+}
+
+// Wrapper for pair of ints
+func (p IntPair) Wrap() UnattributedStatement {
+	return UnattributedStatement{fmt.Sprintf("sum(%v, %v, %v)", p.x, p.y, p.x+p.y)}
+}
+
+func (p IntPair) Identify() Principal {
+	return Principal{"Summer"}
+}
+
+func TestSimpleWrapper(t *testing.T) {
+	handleErr := func(err error) {
+		if err != nil {
+			t.Fatalf("test generated error %v", err)
+			panic(err)
+		}
+	}
+
+	writeErr := EmitWrapperStatement(IntPair{2, 3}, "wrapped_sum.auth_logic")
+	handleErr(writeErr)
+
+	expectedString := "Summer says { sum(2, 3, 5) }"
+	resultBytes, readErr := ioutil.ReadFile("wrapped_sum.auth_logic")
+	handleErr(readErr)
+
+	resultString := string(resultBytes)
+	if expectedString != resultString {
+		t.Fatalf("expected %v got %v", expectedString, resultString)
```

Comment:
The goLang convention is to use "got" and "want", even for the variable names, I think. Also, in error messages "got" should come first (we don't follow this rule everywhere in the code-based, but it will be fixed hopefully soon).

Reply:
fixed